pkgs/by-name/ru/rustlings/package.nix

```
@@ -0,0 +1,55 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+  pkg-config,
+  cargo,
+  rustc,
+  clippy,
+  makeWrapper,
+}:
+let
+  pname = "rustlings";
+  version = "6.2.0";
+in
+rustPlatform.buildRustPackage {
+  inherit pname version;
+  src = fetchFromGitHub {
+    owner = "rust-lang";
+    repo = "rustlings";
+    rev = "v${version}";
+    hash = "sha256-BVmiMIIx8YEO57FO0ZpsCQcim68mn7NHpAOO86dZqlI=";
+  };
+
+  cargoHash = "sha256-ZupwPw/bfeN+s7IWXbY21K/ODXSaYef+IDDpBVCi3Ek=";
+
+  # Disabled test that does not work well in an isolated environment
+  checkFlags = [
+    "--skip=run_compilation_success"
+    "--skip=run_test_success"
+  ];
+
+  nativeBuildInputs = [
+    pkg-config
+    makeWrapper
+  ];
+
+  buildInputs = [
+    cargo
+    rustc
+    clippy
+  ];
+
+  postFixup = ''
+    wrapProgram $out/bin/rustlings --prefix PATH : ${lib.makeBinPath [ cargo ]}
```

Is rustlings dependant on one specific version of the cargo executable or will any do?

If the latter is the case,

```suggestion
    wrapProgram $out/bin/rustlings --suffix PATH : ${lib.makeBinPath [ cargo ]}
```

would be better as it'd otherwise violate the principle of least surprise.